github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17460708?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17460708?buildTab=artifacts#meta) on refs/heads/master @ [d844f6f3be87](https://github.com/cockroachdb/pebble/commits/d844f6f3be87):


```
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 029492
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 029493
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 029494
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 030172
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 030173
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 030174
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 030175
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031280
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031281
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031282
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031283
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031589
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031590
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031591
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031592
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031852
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031853
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031854
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 031855
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 032878
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 032879
        // 06:56:58.440 INFO: [JOB 8616] sstable deleted 032881
        // 06:56:58.440 INFO: [JOB 8625] sstable deleted 033495
        // 06:56:58.440 INFO: [JOB 8626] sstable deleted 033627
        // 06:56:58.440 INFO: [JOB 8626] sstable deleted 033628
        // 06:56:58.468 INFO: [JOB 8621] compacted(download,rewrite) L0 [026796] (871B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (6.1s total), output rate 0B/s
        // 06:56:59.484 INFO: [JOB 8630] compacting(move) L3 [033610] (1.1KB) Score=12.83 + L4 [] (0B) Score=12.90; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:56:59.485 INFO: [JOB 8631] compacting(download,rewrite) L0 [030009] (871B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:56:59.489 INFO: [JOB 8620] compacted(move) L2 [033629] (1.2KB) Score=105.12 + L3 [] (0B) Score=14.45 -> L3 [033629] (1.2KB), in 0.0s (7.2s total), output rate 262MB/s
        // 06:57:00.565 INFO: [JOB 8632] compacting(default) L0 [015841 016237 016765 017133 017704 018377 018652 019029 019435 019988 020515 021440 022593 022856 023147 023224 023225 023332 023333 023614 023615 023786 023787 024216 024217 024580 024581 024900 024901 025112 025113 025993 025994 026451 026452 026866 026867 027229 027230 027610 027611 027873 027874 029033 029034 029497 029498 030178 030179 031286 031287 031595 031596 031858 031859 032884 032885] (54KB) Score=11.71 + L2 [] (0B) Score=7.83; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:57:00.568 INFO: [JOB 8632] compacting: sstable created 033641
        // 06:57:00.568 INFO: [JOB 8632] compacting: sstable created 033642
        // 06:57:00.569 INFO: [JOB 8628] compacted(move) L3 [033625] (1.2KB) Score=13.72 + L4 [] (0B) Score=12.78 -> L4 [033625] (1.2KB), in 0.0s (3.2s total), output rate 219MB/s
        // 06:57:01.460 INFO: [JOB 8633] compacting(move) L2 [033633] (1.2KB) Score=61.19 + L3 [] (0B) Score=12.32; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:57:01.488 INFO: [JOB 8632] compacted(default) L0 [015841 016237 016765 017133 017704 018377 018652 019029 019435 019988 020515 021440 022593 022856 023147 023224 023225 023332 023333 023614 023615 023786 023787 024216 024217 024580 024581 024900 024901 025112 025113 025993 025994 026451 026452 026866 026867 027229 027230 027610 027611 027873 027874 029033 029034 029497 029498 030178 030179 031286 031287 031595 031596 031858 031859 032884 032885] (54KB) Score=11.71 + L2 [] (0B) Score=7.83 -> L2 [033641 033642] (2.3KB), in 0.9s (0.9s total), output rate 2.6KB/s
        // 06:57:02.500 INFO: [JOB 8632] sstable deleted 025112
        // 06:57:02.501 INFO: [JOB 8635] compacting(default) L5 [022048] (1.1KB) Score=69.65 + L6 [011079] (1.0KB) Score=0.02; OverlappingRatio: Single 0.87, Multi 0.00
        // 06:57:02.501 INFO: [JOB 8635] compacting: sstable created 033643
        // 06:57:02.505 INFO: [JOB 8630] compacted(move) L3 [033610] (1.1KB) Score=12.83 + L4 [] (0B) Score=12.90 -> L4 [033610] (1.1KB), in 0.0s (3.0s total), output rate 238MB/s
        // 06:57:03.482 INFO: [JOB 8636] compacting(default) L3 [033629] (1.2KB) Score=12.67 + L4 [] (0B) Score=13.16; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:57:03.482 INFO: [JOB 8632] sstable deleted 025113
        // 06:57:03.482 INFO: [JOB 8636] compacting: sstable created 033644
        // 06:57:03.486 INFO: [JOB 8635] compacted(default) L5 [022048] (1.1KB) Score=69.65 + L6 [011079] (1.0KB) Score=0.02 -> L6 [033643] (1023B), in 1.0s (1.0s total), output rate 1.0KB/s
        // 06:57:04.450 INFO: [JOB 8637] compacting(move) L3 [033611] (1.1KB) Score=12.16 + L4 [] (0B) Score=13.28; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:57:04.450 INFO: [JOB 8632] sstable deleted 025993
        // 06:57:04.450 INFO: [JOB 8632] sstable deleted 025994
        // 06:57:04.454 INFO: [JOB 8636] compacted(default) L3 [033629] (1.2KB) Score=12.67 + L4 [] (0B) Score=13.16 -> L4 [033644] (1.2KB), in 1.0s (1.0s total), output rate 1.2KB/s
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241026-064531.4672789591406/random-005 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-005 (684.64s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1729925131467357984 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: PEBBLE-288

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17503144?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17503144?buildTab=artifacts#meta) on refs/heads/master @ [405b4e1228d5](https://github.com/cockroachdb/pebble/commits/405b4e1228d5):


```
        // 08:57:13.086 INFO: [JOB 8038] compacted(default) L5 [027217 027229] (1.9KB) Score=50.03 + L6 [022907] (986B) Score=0.03 -> L6 [040103] (891B), in 0.0s (1.6s total), output rate 1.8MB/s
        // 08:57:14.631 INFO: [JOB 8040] compacting(default) L5 [028511] (1.0KB) Score=49.71 + L6 [023854] (1016B) Score=0.03; OverlappingRatio: Single 0.99, Multi 0.00
        // 08:57:14.632 INFO: [JOB 8040] compacting: sstable created 040104
        // 08:57:14.668 INFO: [JOB 8039] compacted(download,rewrite) L0 [023615] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (1.6s total), output rate 0B/s
        // 08:57:16.174 INFO: [JOB 8041] compacting(download,rewrite) L0 [034651] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:16.177 INFO: [JOB 8040] compacted(default) L5 [028511] (1.0KB) Score=49.71 + L6 [023854] (1016B) Score=0.03 -> L6 [040104] (909B), in 0.0s (1.5s total), output rate 1.4MB/s
        // 08:57:17.773 INFO: [JOB 8042] compacting(move) L5 [028409] (1010B) Score=49.56 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:17.810 INFO: [JOB 8041] compacted(download,rewrite) L0 [034651] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (1.6s total), output rate 0B/s
        // 08:57:19.431 INFO: [JOB 8043] compacting(download,rewrite) L0 [026669] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:19.434 INFO: [JOB 8042] compacted(move) L5 [028409] (1010B) Score=49.56 + L6 [] (0B) Score=0.03 -> L6 [028409] (1010B), in 0.0s (1.7s total), output rate 167MB/s
        // 08:57:20.979 INFO: [JOB 8044] compacting(move) L5 [028426] (1018B) Score=49.26 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:21.016 INFO: [JOB 8043] compacted(download,rewrite) L0 [026669] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (1.6s total), output rate 0B/s
        // 08:57:22.522 INFO: [JOB 8045] compacting(download,rewrite) L0 [027320] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:22.522 INFO: [JOB 8045] compacting: sstable created 040105
        // 08:57:22.522 INFO: [JOB 8045] compacting: sstable created 040106
        // 08:57:22.525 INFO: [JOB 8044] compacted(move) L5 [028426] (1018B) Score=49.26 + L6 [] (0B) Score=0.03 -> L6 [028426] (1018B), in 0.0s (1.5s total), output rate 266MB/s
        // 08:57:24.017 INFO: [JOB 8046] compacting(default) multilevel L3 [026644] (1.1KB) Score=84.28 + L4 [] (0B) Score=5.12 + L5 [] (0B) Score=48.95; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:24.018 INFO: [JOB 8046] compacting: sstable created 040107
        // 08:57:24.054 INFO: [JOB 8045] compacted(download,rewrite) L0 [027320] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040105 040106] (1.6KB), in 1.5s (1.5s total), output rate 1.1KB/s
        // 08:57:25.588 INFO: [JOB 8047] compacting(download,rewrite) L0 [035041] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:25.592 INFO: [JOB 8046] compacted(default) multilevel L3 [026644] (1.1KB) Score=84.28 + L4 [] (0B) Score=5.12 + L5 [] (0B) Score=48.95 -> L5 [040107] (1.1KB), in 0.0s (1.6s total), output rate 2.8MB/s
        // 08:57:27.207 INFO: [JOB 8049] compacting(move) L5 [028445] (1018B) Score=49.13 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:27.210 INFO: [JOB 8049] compacted(move) L5 [028445] (1018B) Score=49.13 + L6 [] (0B) Score=0.03 -> L6 [028445] (1018B), in 0.0s (0.0s total), output rate 179MB/s
        // 08:57:28.780 INFO: [JOB 8050] compacting(move) L5 [028470] (1010B) Score=48.83 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:28.817 INFO: [JOB 8047] compacted(download,rewrite) L0 [035041] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (3.2s total), output rate 0B/s
        // 08:57:30.397 INFO: [JOB 8051] compacting(download,rewrite) L0 [025108] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:30.400 INFO: [JOB 8050] compacted(move) L5 [028470] (1010B) Score=48.83 + L6 [] (0B) Score=0.03 -> L6 [028470] (1010B), in 0.0s (1.6s total), output rate 164MB/s
        // 08:57:31.899 INFO: [JOB 8052] compacting(move) L5 [028488] (1018B) Score=48.53 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:31.934 INFO: [JOB 8051] compacted(download,rewrite) L0 [025108] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (1.5s total), output rate 0B/s
        // 08:57:33.481 INFO: [JOB 8053] compacting(download,rewrite) L0 [035175] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:33.484 INFO: [JOB 8052] compacted(move) L5 [028488] (1018B) Score=48.53 + L6 [] (0B) Score=0.03 -> L6 [028488] (1018B), in 0.0s (1.6s total), output rate 170MB/s
        // 08:57:35.113 INFO: [JOB 8054] compacting(default) L5 [027315 027316] (2.0KB) Score=48.23 + L6 [023158] (1008B) Score=0.03; OverlappingRatio: Single 0.50, Multi 0.00
        // 08:57:35.113 INFO: [JOB 8054] compacting: sstable created 040108
        // 08:57:35.149 INFO: [JOB 8053] compacted(download,rewrite) L0 [035175] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (1.7s total), output rate 0B/s
        // 08:57:36.704 INFO: [JOB 8055] compacting(download,rewrite) L0 [031018] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:36.707 INFO: [JOB 8054] compacted(default) L5 [027315 027316] (2.0KB) Score=48.23 + L6 [023158] (1008B) Score=0.03 -> L6 [040108] (895B), in 0.0s (1.6s total), output rate 1.5MB/s
        // 08:57:38.246 INFO: [JOB 8056] compacting(default) L5 [027340 027358] (2.0KB) Score=47.92 + L6 [023160] (1008B) Score=0.03; OverlappingRatio: Single 0.50, Multi 0.00
        // 08:57:38.246 INFO: [JOB 8056] compacting: sstable created 040109
        // 08:57:38.282 INFO: [JOB 8055] compacted(download,rewrite) L0 [031018] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [] (0B), in 0.0s (1.6s total), output rate 0B/s
        // 08:57:39.836 INFO: [JOB 8057] compacting(download,rewrite) L0 [027308] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:39.836 INFO: [JOB 8057] compacting: sstable created 040110
        // 08:57:39.839 INFO: [JOB 8056] compacted(default) L5 [027340 027358] (2.0KB) Score=47.92 + L6 [023160] (1008B) Score=0.03 -> L6 [040109] (895B), in 0.0s (1.6s total), output rate 1.5MB/s
        // 08:57:41.356 INFO: [JOB 8058] compacting(default) L5 [028135 028150 028162] (3.0KB) Score=47.60 + L6 [023261] (1019B) Score=0.03; OverlappingRatio: Single 0.33, Multi 0.00
        // 08:57:41.356 INFO: [JOB 8058] compacting: sstable created 040111
        // 08:57:41.392 INFO: [JOB 8057] compacted(download,rewrite) L0 [027308] (810B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040110] (809B), in 1.5s (1.6s total), output rate 532B/s
        // 08:57:42.950 INFO: [JOB 8059] compacting(download,rewrite) L0 [022725] (810B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 08:57:42.953 INFO: [JOB 8058] compacted(default) L5 [028135 028150 028162] (3.0KB) Score=47.60 + L6 [023261] (1019B) Score=0.03 -> L6 [040111] (909B), in 0.0s (1.6s total), output rate 1.1MB/s
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241030-084856.6433102961433/random-018 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-018 (523.97s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730278136643607701 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17558690?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17558690?buildTab=artifacts#meta) on refs/heads/master @ [71bb6ba29095](https://github.com/cockroachdb/pebble/commits/71bb6ba29095):


```
        iter6.Prev("") // [false] <nil> #812
        iter6.Prev("") // [false] <nil> #818
        iter6.Prev("lhavjdiufvr@14") // [invalid] <nil> #820
        // 04:09:51.568 INFO: [JOB 15] flushed 1 memtable (2.6KB) to L0 [000012] (3.1KB), in 0.0s (0.0s total), output rate 815KB/s
        iter5.Next("") // [true,"ubqjcpd@5","quvqyb",<no range>] <nil> #580
        iter5.Next("") // [false] <nil> #581
        iter5.Next("ubqjcpd@3") // [invalid] <nil> #582
        iter5.SeekPrefixGE("ubqjcpd@5") // [true,"ubqjcpd@5","quvqyb",<no range>] <nil> #583
        iter5.SeekLT("ubqjcpd@9", "") // [false] <nil> #584
        iter5.SeekLT("ubqjcpd@9", "") // [false] <nil> #587
        iter5.SeekGE("vtzh", "") // [false] <nil> #588
        iter5.Next("") // [false] <nil> #589
        iter5.Next("") // [false] <nil> #591
        iter5.NextPrefix() // [false] NextPrefix not permitted with upper bound ubqjcpd@4 #592
        iter5.SetOptions("", "gtqivqys@4", 1 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #593
        iter5.Last() // [true,"eqavgms",<no point>,["eqavgms","gtqivqys")=>{"@2"="lvcvzezw"}*] <nil> #594
        iter5.Prev("") // [false] <nil> #596
        iter5.SetBounds("gtqivqys@4", "gtqivqys@4") // <nil> #597
        iter5.SeekLT("gtqivqys@4", "") // [false] <nil> #598
        iter5.Prev("") // [false] <nil> #599
        iter5.SeekGE("qjfmh", "") // [false] <nil> #601
        iter5.SeekLT("qjfmh", "") // [false] <nil> #602
        iter5.Last() // [false] <nil> #604
        iter5.SeekLT("rahfehkx", "") // [false] <nil> #605
        iter5.SeekGE("ubqjcpd@10", "") // [false] <nil> #606
        iter5.SetBounds("gtqivqys@7", "gtqivqys@7") // <nil> #607
        iter6.SeekPrefixGE("ubqjcpd") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #821
        iter6.First() // [false] <nil> #822
        iter6.Last() // [false] <nil> #823
        iter6.SeekPrefixGE("wccnwdiqsesg") // [false] <nil> #829
        iter6.Last() // [false] <nil> #830
        iter6.Next("") // [false] <nil> #831
        iter6.SeekLT("gtqivqys@11", "") // [false] <nil> #833
        iter6.SetBounds("lwau", "mmqmhtqfsjj@13") // <nil> #836
        iter6.SeekLT("mmqmhtqfsjj@13", "") // [true,"mmqmhtqfsjj","ohexjwdseq",["lwau","mmqmhtqfsjj@13")=>{"@7"="rgsla"}*] <nil> #837
        iter6.Next("") // [false] <nil> #838
        // 04:09:51.572 INFO: [JOB 17] compacting(default) L0 [000006 000009 000008 000012] (6.5KB) Score=0.00 + L6 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        iter5.SeekLT("gtqivqys@7", "") // [false] <nil> #608
        iter5.Next("") // [false] <nil> #609
        iter5.Next("") // [false] <nil> #610
        iter5.SetBounds("", "gtqivqys@7") // <nil> #611
        iter5.SeekGE("ubqjcpd@10", "") // [false] <nil> #612
        iter5.SeekGE("gtqivqys@4", "sntvakle") // [invalid] <nil> #613
        iter6.Prev("") // [true,"mmqmhtqfsjj","ohexjwdseq",["lwau","mmqmhtqfsjj@13")=>{"@7"="rgsla"}*] <nil> #839
        iter6.SeekLT("ubqjcpd@8", "lhavjdiufvr@14") // [valid,"mmqmhtqfsjj","ohexjwdseq",["lwau","mmqmhtqfsjj@13")=>{"@7"="rgsla"}] <nil> #846
        iter6.Prev("") // [true,"lwau",<no point>,["lwau","mmqmhtqfsjj@13")=>{"@7"="rgsla"}] <nil> #847
        iter6.Last() // [true,"mmqmhtqfsjj","ohexjwdseq",["lwau","mmqmhtqfsjj@13")=>{"@7"="rgsla"}] <nil> #852
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241102-040947.0991243029628/random-000 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-000 (0.20s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730520587099936041 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17568475?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17568475?buildTab=artifacts#meta) on refs/heads/master @ [71bb6ba29095](https://github.com/cockroachdb/pebble/commits/71bb6ba29095):


```
        iter4.SeekLT("qique@12", "") // [true,"loceupik",<no point>,["loceupik","nortu@18")=>{""="na","@12"="mddkq"}*] <nil> #406
        iter4.Last() // [true,"loceupik",<no point>,["loceupik","nortu@18")=>{""="na","@12"="mddkq"}] <nil> #408
        iter4.SetBounds("xdqkgai", "xgwmnvuvpcnf") // <nil> #414
        iter4.Prev("") // [false] <nil> #418
        iter4.First() // [true,"xdqkgai@13","tirbpcmq",<no range>] <nil> #423
        iter4.SeekGE("pdutbc@2", "rueqfxktrbao@15") // [invalid] <nil> #425
        iter5.SeekLT("abbfaefkrk@3", "") // [false] <nil> #410
        iter5.SetOptions("rueqfxktrbao@15", "rueqfxktrbao@15", 1 /* key types */, 0, 0, true /* use L6 filters */, "" /* masking suffix */) // <nil> #411
        iter5.SeekLT("vstrgaga@4", "") // [false] <nil> #412
        db1.Set("xgwmnvuvpcnf", "hqkvsogmlt") // <nil> #409
        snap3.Get("abbfaefkrk@13") // ["os"] <nil> #413
        iter3.SetBounds("pdutbc@12", "qique@3") // <nil> #429
        iter3.SeekLT("qique@3", "") // [true,"pdutbc@2","cfw",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}*] <nil> #430
        iter3.Next("") // [false] <nil> #431
        iter3.Next("") // [false] <nil> #432
        iter3.Prev("") // [true,"pdutbc@2","cfw",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}*] <nil> #433
        snap4.Get("xdqkgai@13") // ["tirbpcmq"] <nil> #426
        iter4.InternalNext() // <nil> #427
        iter4.SeekPrefixGE("abbfaefkrk") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #428
        iter4.SeekLT("xdqkgai", "") // [false] <nil> #449
        iter4.InternalNext() // <nil> #460
        iter4.First() // [true,"xdqkgai@13","tirbpcmq",<no range>] <nil> #478
        snap3.Get("abbfaefkrk@2") // [""] pebble: not found #435
        iter6 = iter3.Clone(false, "pdutbc@12", "qique@3", 2 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #438
        iter3.InternalNext() // switching from reverse to forward via internalNext is prohibited #439
        iter3.SeekPrefixGE("loceupik") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #440
        iter3.First() // [true,"pdutbc@12","izh",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}*] <nil> #441
        iter3.First() // [true,"pdutbc@12","izh",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}] <nil> #442
        iter3.SeekGE("abbfaefkrk@9", "sxok") // [valid,"pdutbc@12","izh",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}] <nil> #446
        iter3.SeekGE("nxeo", "") // [true,"pdutbc@12","izh",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}] <nil> #447
        iter3.Prev("") // [false] <nil> #450
        iter3.Next("") // [true,"pdutbc@12","izh",["pdutbc@12","qique@3")=>{"@10"="hmkyfxej"}*] <nil> #459
        iter3.SeekGE("rueqfxktrbao@15", "") // [false] <nil> #461
        iter3.SetBounds("sxok@7", "vstrgaga@10") // <nil> #463
        iter5.First() // [false] <nil> #419
        iter5.Next("") // [false] <nil> #420
        iter3.Last() // [true,"sxok@7",<no point>,["sxok@7","vstrgaga@10")=>{"@10"="hmkyfxej"}*] <nil> #465
        iter5.First() // [false] <nil> #434
        iter3.Next("") // [false] <nil> #466
        iter5.SeekGE("effremqxt@13", "") // [false] <nil> #437
        iter5.Next("") // [false] <nil> #462
        iter5.SetBounds("xouteikii@10", "xouteikii@10") // <nil> #468
        iter5.SeekLT("xouteikii@10", "") // [false] <nil> #469
        iter5.Next("") // [false] <nil> #470
        iter5.Next("") // [false] <nil> #471
        iter5.Prev("") // [false] <nil> #472
        iter5.Next("") // [false] <nil> #473
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241103-044434.0752141356262/standard-027 --try-to-reduce -v
        --- FAIL: TestMeta/execution/standard-027 (0.30s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730609074075719627 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17568485?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17568485?buildTab=artifacts#meta) on refs/heads/master @ [71bb6ba29095](https://github.com/cockroachdb/pebble/commits/71bb6ba29095):


```
        iter26.Prev("") // [false] <nil> #2653
        iter28.SeekGE("jkpmt@107", "") // [true,"jkpmt@107",<no point>,["jkpmt@107","lwov")=>{"@87"="oktnqs","@86"="oktnqs","@75"="oktnqs","@73"="cjblauxityllf","@71"="oktnqs","@65"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@52"="oktnqs","@50"="oktnqs","@48"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2591
        iter26.Next("") // [true,"nthfolwmc",<no point>,["nthfolwmc","nuasoobduwx")=>{"@88"="oktnqs","@73"="cjblauxityllf","@66"="oktnqs","@60"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@50"="oktnqs","@48"="oktnqs","@47"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2654
        iter28.Prev("") // [false] <nil> #2592
        iter28.Prev("") // [false] <nil> #2593
        iter28.Next("") // [true,"jkpmt@107",<no point>,["jkpmt@107","lwov")=>{"@87"="oktnqs","@86"="oktnqs","@75"="oktnqs","@73"="cjblauxityllf","@71"="oktnqs","@65"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@52"="oktnqs","@50"="oktnqs","@48"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2594
        iter28.Prev("") // [false] <nil> #2597
        // 05:21:55.193 INFO: [JOB 262] ingested L0:000488 (873B)
        db1.IngestExternalFiles(external1, "rfyqtqh" /* start */, "rfyqtqhzs" /* end */, "@92" /* syntheticSuffix */, "r" /* syntheticPrefix */) // <nil> #2316
        db1.Get("gygbxwoelkf@4") // [""] pebble: not found #2318
        iter28.SeekGE("vrvh@46", "") // [false] <nil> #2601
        iter28.SetBounds("atkmvoovcgde@7", "ggdxpqdqc@1") // <nil> #2605
        iter28.SeekGE("atkmvoovcgde@7", "") // [true,"atkmvoovcgde@7","zemmjllvynwvfpt",<no range>] <nil> #2606
        iter28.Prev("") // [false] <nil> #2607
        iter28.Next("") // [true,"atkmvoovcgde@7","zemmjllvynwvfpt",<no range>] <nil> #2608
        iter28.Next("") // [true,"atkmvoovcgde@6","h",<no range>] <nil> #2609
        iter28.SetBounds("atkmvoovcgde@7", "atkmvoovcgde@7") // <nil> #2612
        iter26.SeekLT("fyqtqh@8", "") // [false] <nil> #2655
        iter26.Prev("") // [false] <nil> #2657
        iter28.SeekLT("atkmvoovcgde@7", "") // [false] <nil> #2613
        iter28.SeekLT("fyqtqh@12", "") // [false] <nil> #2614
        iter28.SeekPrefixGE("iaoglmiogbf") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #2616
        iter28.Prev("tgttpyosqtj@32") // [invalid] pebble: SeekPrefixGE supplied with key outside of upper bound #2617
        iter28.SeekLT("tgttpyosqtj@70", "") // [false] <nil> #2620
        iter28.SeekLT("ehaxglbd@16", "") // [false] <nil> #2621
        iter28.SeekLT("mgntrujzbbqy@13", "") // [false] <nil> #2622
        iter28.SeekPrefixGE("otfylw@11") // [false] pebble: SeekPrefixGE supplied with key outside of upper bound #2625
        iter26.Last() // [true,"otfylw@75","kcdferaeabzdqb",["otfylw","otfylw@60")=>{"@59"="fgmmukuphbwlwl","@54"="tym","@44"="bqjcinbnysnzagysx","@4"="umk"}*] <nil> #2663
        iter26.InternalNext() // switching from reverse to forward via internalNext is prohibited #2664
        iter28.SeekLT("tojurur@15", "") // [false] <nil> #2627
        iter28.NextPrefix() // [false] NextPrefix not permitted with upper bound atkmvoovcgde@7 #2628
        iter28.Last() // [false] <nil> #2631
        iter28.SetBounds("qhon", "yxippnseoh") // <nil> #2642
        iter26.SeekLT("jkpmt@21", "gygbxwoelkf@43") // [invalid] <nil> #2666
        iter26.SeekGE("fyqtqh@6", "") // [true,"nthfolwmc",<no point>,["nthfolwmc","nuasoobduwx")=>{"@88"="oktnqs","@73"="cjblauxityllf","@66"="oktnqs","@60"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@50"="oktnqs","@48"="oktnqs","@47"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2667
        iter26.Prev("vrvh@16") // [invalid] <nil> #2669
        iter26.SeekLT("hclwahvk@42", "") // [false] <nil> #2671
        iter26.SeekPrefixGE("obduwx") // [true,"obduwx","oauevtstdsxahnjzvjq",["obduwx","obduwx\x00")=>{"@88"="oktnqs","@73"="cjblauxityllf","@66"="oktnqs","@60"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@48"="oktnqs","@47"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2674
        iter26.First() // [true,"nthfolwmc",<no point>,["nthfolwmc","nuasoobduwx")=>{"@88"="oktnqs","@73"="cjblauxityllf","@66"="oktnqs","@60"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@50"="oktnqs","@48"="oktnqs","@47"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2678
        iter26.Next("atkmvoovcgde@6") // [invalid] <nil> #2695
        iter26.SeekGE("ggdxpqdqc@8", "") // [true,"nthfolwmc",<no point>,["nthfolwmc","nuasoobduwx")=>{"@88"="oktnqs","@73"="cjblauxityllf","@66"="oktnqs","@60"="oktnqs","@59"="fgmmukuphbwlwl","@54"="tym","@50"="oktnqs","@48"="oktnqs","@47"="oktnqs","@38"="oktnqs","@15"="oktnqs","@4"="umk"}*] <nil> #2702
        // 05:21:55.211 INFO: [JOB 264] ingested L0:000489 (920B)
        db1.IngestExternalFiles(external0, "iaoglmiogbf" /* start */, "mgntrujzbbqy" /* end */, "@93" /* syntheticSuffix */, "" /* syntheticPrefix */) // <nil> #2330
        db1.Set("gygbxwoelkf@42", "wbkwqozbx") // <nil> #2333
        db1.Set("ehaxglbd@91", "wzm") // <nil> #2336
        iter28.SeekLT("yxippnseoh", "") // [true,"yveonpfygzuumaoj",<no point>,["yveonpfygzuumaoj","yveootfylw")=>{"@44"="oktnqs"}*] <nil> #2643
        db1.Get("atkmvoovcgde@6") // ["h"] <nil> #2339
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241103-052125.159585152575/random-022 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-022 (3.87s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730611285159663832 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17578361?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17578361?buildTab=artifacts#meta) on refs/heads/master @ [71bb6ba29095](https://github.com/cockroachdb/pebble/commits/71bb6ba29095):


```
        // 06:45:08.589 INFO: [JOB 2232] compacting: sstable created 040615
        // 06:45:08.590 INFO: [JOB 2231] compacted(move) L4 [031963] (1.1KB) Score=12.68 + L5 [] (0B) Score=0.76 -> L5 [031963] (1.1KB), in 0.0s (1.6s total), output rate 240MB/s
        // 06:45:10.127 INFO: [JOB 2233] compacting(move) L3 [036649] (917B) Score=13.55 + L4 [] (0B) Score=12.54; OverlappingRatio: Single 0.00, Multi 0.38
        // 06:45:10.139 INFO: [JOB 2232] compacted(download,rewrite) L0 [035859] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040615] (1014B), in 1.5s (1.6s total), output rate 659B/s
        // 06:45:11.670 INFO: [JOB 2234] compacting(download,rewrite) L0 [036635] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:11.670 INFO: [JOB 2234] compacting: sstable created 040616
        // 06:45:11.672 INFO: [JOB 2233] compacted(move) L3 [036649] (917B) Score=13.55 + L4 [] (0B) Score=12.54 -> L4 [036649] (917B), in 0.0s (1.5s total), output rate 205MB/s
        // 06:45:13.198 INFO: [JOB 2235] compacting(move) L4 [026076] (1.1KB) Score=12.57 + L5 [] (0B) Score=0.77; OverlappingRatio: Single 0.00, Multi 0.87
        // 06:45:13.214 INFO: [JOB 2234] compacted(download,rewrite) L0 [036635] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040616] (1014B), in 1.5s (1.5s total), output rate 663B/s
        // 06:45:14.733 INFO: [JOB 2236] compacting(download,rewrite) L0 [036691] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:14.733 INFO: [JOB 2236] compacting: sstable created 040617
        // 06:45:14.735 INFO: [JOB 2235] compacted(move) L4 [026076] (1.1KB) Score=12.57 + L5 [] (0B) Score=0.77 -> L5 [026076] (1.1KB), in 0.0s (1.5s total), output rate 87MB/s
        // 06:45:16.326 INFO: [JOB 2237] compacting(move) L3 [036650] (925B) Score=13.52 + L4 [] (0B) Score=12.49; OverlappingRatio: Single 0.00, Multi 0.29
        // 06:45:16.338 INFO: [JOB 2236] compacted(download,rewrite) L0 [036691] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040617] (1014B), in 1.6s (1.6s total), output rate 636B/s
        // 06:45:17.907 INFO: [JOB 2238] compacting(download,rewrite) L0 [036758] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:17.907 INFO: [JOB 2238] compacting: sstable created 040618
        // 06:45:17.908 INFO: [JOB 2237] compacted(move) L3 [036650] (925B) Score=13.52 + L4 [] (0B) Score=12.49 -> L4 [036650] (925B), in 0.0s (1.6s total), output rate 178MB/s
        // 06:45:19.513 INFO: [JOB 2239] compacting(move) L4 [026077] (1.5KB) Score=12.52 + L5 [] (0B) Score=0.77; OverlappingRatio: Single 0.00, Multi 0.51
        // 06:45:19.524 INFO: [JOB 2238] compacted(download,rewrite) L0 [036758] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040618] (1014B), in 1.6s (1.6s total), output rate 631B/s
        // 06:45:21.027 INFO: [JOB 2240] compacting(download,rewrite) L0 [037039] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:21.027 INFO: [JOB 2240] compacting: sstable created 040619
        // 06:45:21.029 INFO: [JOB 2239] compacted(move) L4 [026077] (1.5KB) Score=12.52 + L5 [] (0B) Score=0.77 -> L5 [026077] (1.5KB), in 0.0s (1.5s total), output rate 327MB/s
        // 06:45:22.584 INFO: [JOB 2241] compacting(move) multilevel L3 [036651] (925B) Score=13.51 + L4 [] (0B) Score=12.40 + L5 [] (0B) Score=0.78; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:22.596 INFO: [JOB 2240] compacted(download,rewrite) L0 [037039] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040619] (1014B), in 1.6s (1.6s total), output rate 651B/s
        // 06:45:24.134 INFO: [JOB 2242] compacting(download,rewrite) L0 [037042] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:24.134 INFO: [JOB 2242] compacting: sstable created 040620
        // 06:45:24.135 INFO: [JOB 2241] compacted(move) multilevel L3 [036651] (925B) Score=13.51 + L4 [] (0B) Score=12.40 + L5 [] (0B) Score=0.78 -> L5 [036651] (925B), in 0.0s (1.6s total), output rate 218MB/s
        // 06:45:25.669 INFO: [JOB 2243] compacting(move) multilevel L3 [036652] (913B) Score=13.42 + L4 [] (0B) Score=12.37 + L5 [] (0B) Score=0.78; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:25.681 INFO: [JOB 2242] compacted(download,rewrite) L0 [037042] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040620] (1014B), in 1.5s (1.5s total), output rate 660B/s
        // 06:45:27.226 INFO: [JOB 2244] compacting(download,rewrite) L0 [037045] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:27.226 INFO: [JOB 2244] compacting: sstable created 040621
        // 06:45:27.227 INFO: [JOB 2243] compacted(move) multilevel L3 [036652] (913B) Score=13.42 + L4 [] (0B) Score=12.37 + L5 [] (0B) Score=0.78 -> L5 [036652] (913B), in 0.0s (1.6s total), output rate 224MB/s
        // 06:45:28.795 INFO: [JOB 2245] compacting(move) L3 [036662] (921B) Score=13.33 + L4 [] (0B) Score=12.34; OverlappingRatio: Single 0.00, Multi 1.21
        // 06:45:28.807 INFO: [JOB 2244] compacted(download,rewrite) L0 [037045] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040621] (1014B), in 1.6s (1.6s total), output rate 645B/s
        // 06:45:30.363 INFO: [JOB 2246] compacting(download,rewrite) L0 [037351] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:30.363 INFO: [JOB 2246] compacting: sstable created 040622
        // 06:45:30.365 INFO: [JOB 2245] compacted(move) L3 [036662] (921B) Score=13.33 + L4 [] (0B) Score=12.34 -> L4 [036662] (921B), in 0.0s (1.6s total), output rate 125MB/s
        // 06:45:31.843 INFO: [JOB 2247] compacting(move) L4 [029707] (1.1KB) Score=12.38 + L5 [] (0B) Score=0.79; OverlappingRatio: Single 0.00, Multi 0.40
        // 06:45:31.855 INFO: [JOB 2246] compacted(download,rewrite) L0 [037351] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040622] (1014B), in 1.5s (1.5s total), output rate 684B/s
        // 06:45:33.360 INFO: [JOB 2248] compacting(download,rewrite) L0 [037803] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:33.360 INFO: [JOB 2248] compacting: sstable created 040623
        // 06:45:33.361 INFO: [JOB 2247] compacted(move) L4 [029707] (1.1KB) Score=12.38 + L5 [] (0B) Score=0.79 -> L5 [029707] (1.1KB), in 0.0s (1.5s total), output rate 183MB/s
        // 06:45:34.899 INFO: [JOB 2249] compacting(move) multilevel L1 [036623] (1.2KB) Score=17.04 + L2 [] (0B) Score=3.29 + L3 [] (0B) Score=13.30; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:34.911 INFO: [JOB 2248] compacted(download,rewrite) L0 [037803] (1006B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [040623] (1014B), in 1.5s (1.6s total), output rate 658B/s
        // 06:45:36.445 INFO: [JOB 2250] compacting(download,rewrite) L0 [038281] (1006B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 06:45:36.446 INFO: [JOB 2250] compacting: sstable created 040624
        // 06:45:36.447 INFO: [JOB 2249] compacted(move) multilevel L1 [036623] (1.2KB) Score=17.04 + L2 [] (0B) Score=3.29 + L3 [] (0B) Score=13.30 -> L3 [036623] (1.2KB), in 0.0s (1.5s total), output rate 304MB/s
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241104-063534.4103198259270/random-006 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-006 (577.95s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730702134410362756 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17595924?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/17595924?buildTab=artifacts#meta) on refs/heads/master @ [1af22dc322eb](https://github.com/cockroachdb/pebble/commits/1af22dc322eb):


```
        iter4.First() // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #575
        iter4.Prev("") // [false] <nil> #576
        iter4.Last() // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #577
        iter4.SeekLT("xnujiohzo@7", "wvpsuwrq") // [invalid] <nil> #578
        iter4.First() // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #579
        iter4.Prev("") // [false] <nil> #580
        iter4.SeekLT("gdujwkweb@14", "") // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #581
        iter4.Last() // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}] <nil> #582
        iter4.SeekGE("wvpsuwrq@12", "xnujiohzo") // [invalid] <nil> #583
        iter4.Next("") // [false] <nil> #584
        iter4.Next("") // [false] <nil> #585
        iter4.SeekLT("hevobxyuuxxp", "") // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #586
        iter4.SeekGE("esyp@11", "esyp@9") // [invalid] <nil> #587
        iter4.SeekPrefixGE("esyp") // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #588
        iter4.Prev("") // [false] pebble: unsupported reverse prefix iteration #589
        iter4.NextPrefix() // [false] NextPrefix not permitted with upper bound esyp@15 #590
        iter4.SeekGE("oahn@17", "") // [false] <nil> #591
        iter4.Last() // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}*] <nil> #592
        iter4.First() // [true,"esyp",<no point>,["esyp","esyp@15")=>{"@5"="mxidi"}] <nil> #595
        db1.Get("ipqynfce") // ["zhys"] <nil> #544
        iter4.SetBounds("jtsmdrlcxh@13", "wvpsuwrq@3") // <nil> #596
        snap7.Get("pgjtavbph") // ["lmt"] <nil> #547
        db1.Get("yejherkipf") // [""] pebble: not found #550
        db1.Set("wvpsuwrq@3", "xqulxalhsp") // <nil> #551
        db1.SingleDelete("gdujwkweb", false /* maybeReplaceDelete */) // <nil> #571
        snap7.Get("oahn") // [""] pebble: not found #572
        snap7.Get("esyp@15") // ["sesnbu"] <nil> #574
        db1.Get("mybrfettv") // [""] pebble: not found #593
        db1.Set("hevobxyuuxxp@16", "xvedobhkcstnv") // <nil> #594
        db1.Get("gdujwkweb@14") // ["gthahi"] <nil> #611
        iter4.SeekLT("wvpsuwrq@3", "") // [true,"wvpsuwrq@10","pmodpoctjzoqkkdgrxq",<no range>] <nil> #597
        iter4.Next("") // [false] <nil> #598
        iter4.Next("") // [false] <nil> #599
        iter4.Prev("") // [true,"wvpsuwrq@10","pmodpoctjzoqkkdgrxq",<no range>] <nil> #600
        iter4.First() // [true,"jtsmdrlcxh@13",<no point>,["jtsmdrlcxh@13","mybrfettv")=>{"@15"="vlqumprtougrv","@5"="mxidi"}*] <nil> #601
        db1.Delete("esyp@3") // <nil> #615
        iter4.SeekPrefixGE("oahn@12") // [true,"oahn@12","souzsdmanw",["oahn","oahn\x00")=>{"@15"="vlqumprtougrv"}*] <nil> #602
        iter4.SetBounds("xnujiohzo", "yejherkipf") // <nil> #603
        iter4.First() // [false] <nil> #604
        iter4.First() // [false] <nil> #605
        snap7.Get("oahn@14") // ["jyfgwswvzcizndj"] <nil> #618
        iter4.SeekGE("xnujiohzo@3", "") // [false] <nil> #606
        iter4.SetBounds("eagkhwpnd", "ipqynfce@15") // <nil> #607
        db1.LogData("fcq") // <nil> #623
        db1.Set("ipqynfce@17", "knekaoiwk") // <nil> #624
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241105-052154.6331508676450/standard-018 --try-to-reduce -v
        --- FAIL: TestMeta/execution/standard-018 (0.21s)
=== RUN   TestMeta/execution
    --- FAIL: TestMeta/execution (0.00s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730784114633139917 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17613086?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/17613086?buildTab=artifacts#meta) on refs/heads/master @ [2da617a0a886](https://github.com/cockroachdb/pebble/commits/2da617a0a886):


```
        // 05:37:33.728 INFO: [JOB 1200] compacted(default) L5 [028600] (1.0KB) Score=35.52 + L6 [023882] (985B) Score=0.03 -> L6 [031618 031619] (1.9KB), in 0.0s (1.0s total), output rate 524KB/s
        // 05:37:34.789 INFO: [JOB 1203] compacting(default) L4 [027363] (1.0KB) Score=48.48 + L5 [021954] (985B) Score=35.18; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:34.789 INFO: [JOB 1203] compacting: sstable created 031621
        // 05:37:34.803 INFO: [JOB 1202] compacted(download,copy) L0 [029242] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031620] (724B), in 0.0s (1.1s total), output rate 227KB/s
        // 05:37:35.832 INFO: [JOB 1205] compacting(download,copy) L0 [019694] (724B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:35.834 INFO: [JOB 1203] compacted(default) L4 [027363] (1.0KB) Score=48.48 + L5 [021954] (985B) Score=35.18 -> L5 [031621] (1.0KB), in 0.0s (1.0s total), output rate 291KB/s
        // 05:37:36.871 INFO: [JOB 1207] compacting(default) L4 [027370] (1.0KB) Score=47.99 + L5 [025351] (993B) Score=35.19; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:36.871 INFO: [JOB 1207] compacting: sstable created 031623
        // 05:37:36.886 INFO: [JOB 1205] compacted(download,copy) L0 [019694] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031622] (724B), in 1.0s (1.1s total), output rate 697B/s
        // 05:37:37.930 INFO: [JOB 1209] compacting(download,copy) L0 [021291] (853B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:37.932 INFO: [JOB 1207] compacted(default) L4 [027370] (1.0KB) Score=47.99 + L5 [025351] (993B) Score=35.19 -> L5 [031623] (1.0KB), in 0.0s (1.1s total), output rate 268KB/s
        // 05:37:38.901 INFO: [JOB 1210] compacting(default) L5 [023363] (1001B) Score=35.20 + L6 [031619] (931B) Score=0.03; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:38.916 INFO: [JOB 1209] compacted(download,copy) L0 [021291] (853B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031624] (799B), in 1.0s (1.0s total), output rate 823B/s
        // 05:37:39.969 INFO: [JOB 1213] compacting(download,copy) L0 [025970] (724B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:39.972 INFO: [JOB 1210] compacted(default) L5 [023363] (1001B) Score=35.20 + L6 [031619] (931B) Score=0.03 -> L6 [] (0B), in 0.0s (1.1s total), output rate 0B/s
        // 05:37:40.971 INFO: [JOB 1214] compacting(move) L4 [027409] (1.0KB) Score=47.75 + L5 [] (0B) Score=35.15; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:40.986 INFO: [JOB 1213] compacted(download,copy) L0 [025970] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031625] (724B), in 1.0s (1.0s total), output rate 722B/s
        // 05:37:42.022 INFO: [JOB 1216] compacting(download,copy) L0 [027337] (724B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:42.022 INFO: [JOB 1213] sstable deleted 031619
        // 05:37:42.024 INFO: [JOB 1214] compacted(move) L4 [027409] (1.0KB) Score=47.75 + L5 [] (0B) Score=35.15 -> L5 [027409] (1.0KB), in 0.0s (1.1s total), output rate 231MB/s
        // 05:37:43.062 INFO: [JOB 1217] compacting(move) L5 [027409] (1.0KB) Score=35.54 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:43.076 INFO: [JOB 1216] compacted(download,copy) L0 [027337] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031626] (724B), in 1.0s (1.1s total), output rate 696B/s
        // 05:37:44.090 INFO: [JOB 1219] compacting(download,copy) L0 [029432] (724B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:44.092 INFO: [JOB 1217] compacted(move) L5 [027409] (1.0KB) Score=35.54 + L6 [] (0B) Score=0.03 -> L6 [027409] (1.0KB), in 0.0s (1.0s total), output rate 217MB/s
        // 05:37:45.152 INFO: [JOB 1220] compacting(default) L4 [027406] (1.0KB) Score=47.27 + L5 [023173] (985B) Score=34.99; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:45.152 INFO: [JOB 1220] compacting: sstable created 031628
        // 05:37:45.168 INFO: [JOB 1219] compacted(download,copy) L0 [029432] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031627] (724B), in 1.1s (1.1s total), output rate 681B/s
        // 05:37:46.138 INFO: [JOB 1222] compacting(download,copy) L0 [029871] (724B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:46.140 INFO: [JOB 1220] compacted(default) L4 [027406] (1.0KB) Score=47.27 + L5 [023173] (985B) Score=34.99 -> L5 [031628] (1.0KB), in 0.0s (1.0s total), output rate 308KB/s
        // 05:37:47.168 INFO: [JOB 1223] compacting(default) L5 [028631] (1.1KB) Score=35.00 + L6 [023824] (1007B) Score=0.03; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:47.168 INFO: [JOB 1223] compacting: sstable created 031630
        // 05:37:47.168 INFO: [JOB 1223] compacting: sstable created 031631
        // 05:37:47.183 INFO: [JOB 1222] compacted(download,copy) L0 [029871] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031629] (724B), in 1.0s (1.0s total), output rate 702B/s
        // 05:37:48.242 INFO: [JOB 1226] compacting(download,copy) L0 [024049] (724B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:48.247 INFO: [JOB 1223] compacted(default) L5 [028631] (1.1KB) Score=35.00 + L6 [023824] (1007B) Score=0.03 -> L6 [031630 031631] (1.9KB), in 0.0s (1.1s total), output rate 524KB/s
        // 05:37:49.240 INFO: [JOB 1227] compacting(default) L4 [027407] (1.0KB) Score=47.05 + L5 [023253] (985B) Score=34.67; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:49.241 INFO: [JOB 1227] compacting: sstable created 031633
        // 05:37:49.257 INFO: [JOB 1226] compacted(download,copy) L0 [024049] (724B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031632] (724B), in 0.0s (1.0s total), output rate 234KB/s
        // 05:37:50.275 INFO: [JOB 1229] compacting(download,copy) L0 [017707] (853B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:50.278 INFO: [JOB 1227] compacted(default) L4 [027407] (1.0KB) Score=47.05 + L5 [023253] (985B) Score=34.67 -> L5 [031633] (1.0KB), in 0.0s (1.0s total), output rate 225KB/s
        // 05:37:51.282 INFO: [JOB 1231] compacting(default) L4 [027424] (1.0KB) Score=46.56 + L5 [025356] (1001B) Score=34.68; OverlappingRatio: Single 0.93, Multi 0.00
        // 05:37:51.283 INFO: [JOB 1231] compacting: sstable created 031635
        // 05:37:51.298 INFO: [JOB 1229] compacted(download,copy) L0 [017707] (853B) Score=0.00 + L0 [] (0B) Score=0.00 -> L0 [031634] (799B), in 1.0s (1.0s total), output rate 793B/s
        // 05:37:52.347 INFO: [JOB 1233] compacting(download,copy) L0 [029437] (853B) Score=0.00 + L0 [] (0B) Score=0.00; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:52.349 INFO: [JOB 1231] compacted(default) L4 [027424] (1.0KB) Score=46.56 + L5 [025356] (1001B) Score=34.68 -> L5 [031635] (1.0KB), in 0.0s (1.1s total), output rate 282KB/s
        // 05:37:53.411 INFO: [JOB 1235] compacting(move) L5 [028654] (1.0KB) Score=34.69 + L6 [] (0B) Score=0.03; OverlappingRatio: Single 0.00, Multi 0.00
        // 05:37:53.413 INFO: [JOB 1235] compacted(move) L5 [028654] (1.0KB) Score=34.69 + L6 [] (0B) Score=0.03 -> L6 [028654] (1.0KB), in 0.0s (0.0s total), output rate 214MB/s
        
        To reduce:  go test ./internal/metamorphic -tags invariants -run 'TestMeta$' --run-dir /artifacts/meta/241106-053141.8214141332955/random-014 --try-to-reduce -v
        --- FAIL: TestMeta/execution/random-014 (372.76s)
```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run 'TestMeta$' ./internal/metamorphic -seed 1730871101821123941 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/internal/metamorphic/metaflags/meta_flags.go
+++ b/internal/metamorphic/metaflags/meta_flags.go
@@ -76,7 +76,7 @@ func initCommonFlags() *CommonFlags {
 
 	flag.IntVar(&c.NumInstances, "num-instances", 1, "number of pebble instances to create (default: 1)")
 
-	defaultOpTimeout := 1 * time.Minute
+	defaultOpTimeout := 2 * time.Minute
 	if invariants.RaceEnabled {
 		defaultOpTimeout *= 5
 	}

--- a/metamorphic/test.go
+++ b/metamorphic/test.go
@@ -423,10 +423,12 @@ func (t *Test) runOp(idx int, h historyRecorder) {
 	if t.opTimeout > 0 {
 		opTimeout := t.opTimeout
 		switch op.(type) {
-		case *compactOp, *downloadOp, *newSnapshotOp, *ingestOp, *ingestAndExciseOp, *ingestExternalFilesOp:
+		case *compactOp, *newSnapshotOp, *ingestOp, *ingestAndExciseOp, *ingestExternalFilesOp:
 			// These ops can be very slow, especially if we end up with many tiny
 			// tables. Bump up the timout by a factor.
 			opTimeout *= 4
+		case *downloadOp:
+			opTimeout *= 8
 		}
 		timer = time.AfterFunc(opTimeout, func() {
 			panic(fmt.Sprintf("operation took longer than %s: %s", opTimeout, op.String()))

